thetis/callback.py

```diff
@@ -602,9 +602,10 @@ class TimeSeriesCallback2D(DiagnosticCallback):
 
     @PETSc.Log.EventDecorator("thetis.TimeSeriesCallback2D.__init__")
     def __init__(self, solver_obj, fieldnames, x, y,
-                 location_name,
+                 location_name, z=None,
```

Comment:
Just a general comment unrelated to this PR. Why do we have both a `TimeSeriesCallback2D` and a `DetectorsCallback`? They seem to be very similar and we might be duplicating effort, such as the changes here...

Reply:
This is a good point - `DetectorsCallback` predates the `TimeSeriesCallback2d`. I guess we could keep the latter as it is more abstract, though `DetectorsCallback` is a bit more intuitive for new users in my view ...

Reply:
Yes, I think the design difference is that `DetectorsCallback` stores all the time series in a single hdf5 file while `TimeSeriesCallback` creates separate files. In the past I've found it easier to work with separate files in post processing. But I think we'll need to revisit this issue if/when we generalize the output scheduling, metadata (e.g. storing time as calendar dates) and formats (e.g. produce standard netCDF time series files).